File: app/constants/project_links.py
"""Project link entity types."""

PROJECT_LINK_TYPES = frozenset(
    {"asset", "goal", "document", "finance_account", "task", "decision"}
)

PROJECT_LINK_LABELS = {
    "asset": "Eiendel",
    "goal": "Mål",
    "document": "Dokument",
    "finance_account": "Finanskonto",
    "task": "Oppgave",
    "decision": "Beslutning",
}

ENTITY_COLLECTIONS = {
    "asset": "assets",
    "goal": "goals",
    "document": "documents",
    "finance_account": "finance_accounts",
    "task": "tasks",
    "decision": "decisions",
}

ENTITY_DETAIL_PATHS = {
    "asset": "/assets/{id}",
    "goal": "/goals/{id}",
    "document": "/documents",
    "finance_account": "/finance",
    "task": "/tasks",
    "decision": "/decisions",
}

File: web/src/lib/project-links.ts
export const PROJECT_LINK_TYPES = [
  { value: "asset", label: "Eiendel", path: "/assets" },
  { value: "goal", label: "Mål", path: "/goals" },
  { value: "task", label: "Oppgave", path: "/tasks" },
  { value: "document", label: "Dokument", path: "/documents" },
  { value: "finance_account", label: "Finanskonto", path: "/finance" },
  { value: "decision", label: "Beslutning", path: "/decisions" },
] as const;

export type ProjectLinkType = (typeof PROJECT_LINK_TYPES)[number]["value"];

export function projectLinkTypeLabel(type: unknown): string {
  const match = PROJECT_LINK_TYPES.find((item) => item.value === type);
  return match?.label ?? String(type ?? "Kobling");
}

export function entityRecordLabel(record: Record<string, unknown>): string {
  for (const field of ["name", "title", "filename"]) {
    if (record[field]) return String(record[field]);
  }
  return String(record.id ?? "Ukjent");
}
